test/partiql-randomized-tests/src/test/kotlin/org/partiql/lang/randomized/eval/EvaluatingCompilerDateTimeRandomizedTests.kt

```diff
@@ -1,178 +1,177 @@
+                withPrecision = true,
+                withTimezone = true
+            )
+        }
+    }
+
+    data class TimeForValidation(
+        val hour: Int,
+        val minute: Int,
+        val second: Int,
+        val nano: Int = 0,
+        val precision: Int,
+        val tz_minutes: Int? = null
+    ) {
+        fun expectedTimeString(withTimeZone: Boolean): String {
+            val timezoneMinutes = when (withTimeZone) {
+                true -> tz_minutes ?: (ZoneOffset.UTC.totalSeconds / SECONDS_PER_MINUTE)
+                else -> null
+            }
+            return TimeForValidation(hour, minute, second, nano, precision, timezoneMinutes).toString()
+        }
+
+        override fun toString(): String {
+            val hourStr = hour.toString().padStart(2, '0')
+            val minStr = minute.toString().padStart(2, '0')
+            val secStr = second.toString().padStart(2, '0')
+            val nanoStr = nano.toString().padStart(9, '0')
+            val timezoneStr = tz_minutes?.let {
+                "" +
+                    (if (it >= 0) "+" else "-") +
+                    (it.absoluteValue / 60).toString().padStart(2, '0') +
+                    ":" +
+                    (it.absoluteValue % 60).toString().padStart(2, '0')
+            } ?: ""
+            return "$hourStr:$minStr:$secStr.$nanoStr$timezoneStr"
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("randomTimes")
+    fun testRandomTimes(time: TimeForValidation) {
+        val query = "TIME '$time'"
+        val expected = "TIME '${time.expectedTimeString(withTimeZone = false)}'"
+        runEvaluatorTestCase(
+            query = query,
+            expectedResult = expected
+        )
+    }
+
+    @ParameterizedTest
+    @MethodSource("randomTimesWithPrecision")
+    fun testRandomTimesWithPrecision(time: TimeForValidation) {
+        val query = "TIME (${time.precision}) '$time'"
+        val expected = "TIME '${time.expectedTimeString(withTimeZone = false)}'"
+        runEvaluatorTestCase(
+            query = query,
+            expectedResult = expected
+        )
+    }
+
+    @ParameterizedTest
+    @MethodSource("randomTimesWithTimezone")
+    fun testRandomTimesWithTimezone(time: TimeForValidation) {
+        val query = "TIME WITH TIME ZONE '$time'"
+        val expected = "TIME WITH TIME ZONE '${time.expectedTimeString(withTimeZone = true)}'"
+        runEvaluatorTestCase(
+            query = query,
+            expectedResult = expected
+        )
+    }
+
+    @ParameterizedTest
+    @MethodSource("randomTimesWithPrecisionAndTimezone")
+    fun testRandomTimesWithPrecisionAndTimezone(time: TimeForValidation) {
+        val query = "TIME (${time.precision}) WITH TIME ZONE '$time'"
+        val expected = "TIME WITH TIME ZONE '${time.expectedTimeString(withTimeZone = true)}'"
+        runEvaluatorTestCase(
+            query = query,
+            expectedResult = expected
+        )
+    }
+
+    companion object {
```

Comment:
(self-review) had to rewrite a big chunk of this test file since the prior modeling was pretty confusing, especially for how it reused test cases. We could clean up these tests further, but given the time constraints, didn't want to spend too long.